src/parser.rs

```diff
@@ -1330,6 +1332,10 @@ pub fn parse_function(
                     )));
                 }
 
+                if error.is_some() {
```

Comment:
can you move these two if blocks up and into the previous else?

```
trace!("ERROR: incomplete function");
error = error.or(Some(JaktError::ParserError(
    "incomplete function".to_string(),
    tokens[*index - 1].span,
)));
return (ParsedFunction::new(FunctionLinkage::Internal), error);
```

Reply:
If that's true, I'd probably rather we copy it so that the error is closer to the error occurring. When we have distance like this, it's possible that a later refactor will break the invariant that we know the kind of error that just occurred. If we move them so they're right beside when the error occurs, this is less likely to happen.